### PR TITLE
ci: Run `vmtests` before `go-build` tests

### PR DESCRIPTION
**Pull request by Namanl2001**
### Why?

The issue explained clearly in the description: https://github.com/parca-dev/parca-agent/issues/1707

### What?

a) moved the `vmtests` job above `go-build-test`
b) removed the following steps entirely as they are not required now
- `Upload initramfs`
- `Download previously generated initramfs`

### Test Plan

Tested in fork: https://github.com/Namanl2001/parca-agent/actions/runs/5424646940
GH action link: https://github.com/parca-dev/parca-agent/actions/runs/5426919153

cc: @javierhonduco @kakkoyun @Sylfrena

### CODE REVIEW

**Review by javierhonduco — APPROVED** [review]
LGTM, let's wait for another maintainer to approve it too, though!

**Review by brancz — APPROVED** [review]
Looks good to me on green! 🍏

**Comment by brancz** [comment]
Thank you very much for the contribution!

**Pull request merged**
